packages/contracts-bedrock/src/cannon/libraries/MIPS64Instructions.sol

```diff
@@ -595,8 +595,8 @@ library MIPS64Instructions {
     function signExtend(uint64 _dat, uint64 _idx) internal pure returns (uint64 out_) {
         unchecked {
             bool isSigned = (_dat >> (_idx - 1)) & 1 != 0;
-            uint256 signed = ((1 << (arch.WORD_SIZE - _idx)) - 1) << _idx;
             uint256 mask = (1 << _idx) - 1;
+            uint256 signed = ~mask;
```

Comment:
This should also be updated in:
- The corresponding [go code](https://github.com/ethereum-optimism/optimism/blob/8c040995f3c0630f21ae7fcc56b3621c75f9445f/cannon/mipsevm/exec/mips_instructions.go#L469-L478)
- The corresponding 32-bit logic: [MIPSInstructions](https://github.com/ethereum-optimism/optimism/blob/8c040995f3c0630f21ae7fcc56b3621c75f9445f/packages/contracts-bedrock/src/cannon/libraries/MIPSInstructions.sol#L444-L452)

And the [contract versions](https://github.com/ethereum-optimism/optimism/blob/8c040995f3c0630f21ae7fcc56b3621c75f9445f/packages/contracts-bedrock/src/cannon/MIPS64.sol#L65-L67) should be bumped in the contracts that use these helpers: MIPS.sol, MIPS2.sol, MIPS64.sol.

Reply:
You'll have to merge in `develop` - we have removed  `MIPS.sol`, `MIPS2.sol`, `MIPSInstructions.sol` since this PR was opened.